day-26-challenge/solution.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+module.exports = exports = {};
+
+exports.map = function (array, callback) {
+  if (!Array.isArray(array)) throw new Error('expected array');
+  let returnArray = [];
+  for (let i = 0; i < array.length; i++) {
+    returnArray.push(callback(array[i]));
+  }
+  return returnArray;
+};
+
+exports.filter = function (array, callback) {
+  if (!Array.isArray(array)) throw new Error('expected array');
+  let returnArray = [];
+  for (let i = 0; i < array.length; i++) {
+    if (callback(array[i])) {
+      returnArray.push(array[i]);
+    }
+  }
+  return returnArray
+};
+
+exports.reduce = function (array, acccumulator, start) {
```

Comment:
Careful with your notation here. Generally the callback is formulated as (accumulater, currentValue) and start would be initialValue